Organize repo

adamaltman opened the pull request:
To do...
- [x] implement new router
- [ ] de-couple models sql data from entity
- [ ] implement aura views

@brandonsavage

adamaltman committed:
Change Vagrantfile setup...

- homestead error; reverted to rasmus' php7dev box
- configured nginx and mysql

adamaltman committed:
wip: organizing repo, introducing di (partially)

adamaltman committed:
move sql to models, finish DI setup

adamaltman committed:
Request object, and usage of it in Controller classes

adamaltman committed:
create DataStore interface and Mysql implementation, and integrate all models to use new code

adamaltman committed:
clean out unused MasterController code

brandonsavage commented on diconfig.php:
```diff
@@ -0,0 +1,59 @@
+    ],
+];
+
+$di->params[Masterclass\Db\Mysql::class] = [
```
Good use of ::class.

brandonsavage commented on diconfig.php:
```diff
@@ -0,0 +1,59 @@
+    'pdo' => $di->lazyNew('PDO'),
+];
+
+$di->params[Masterclass\Request::class] = [
```
For services like this I like to set an easy-to-remember key, like:
`
$di->set('request', $di->lazyNew('Masterclass\Request');
`

adamaltman replied:
Oh, thank you.

brandonsavage commented on src/Controller/Comment.php:
```diff
@@ -0,0 +1,36 @@
+    public function create()
+    {
+        if (!isset($_SESSION['AUTHENTICATED'])) {
+            // @todo: die, redirect, exit... what is desired functionality?
```
I think that's what we call "a bug." Desired functionality should be to redirect.

brandonsavage commented on src/Db/Apc.php:
```diff
@@ -0,0 +1,37 @@
+
+        return $value;
+    }
+    public function fetchAll($sql, array $args = [])
```
Don't forget to fill these in!

adamaltman replied:
Yes, I don't think I have apcu.  Is it a php module?

brandonsavage commented:
This is a good starting point. I'll give it a more thorough check in the morning. Note that you created the PR with the destination of my repo, not your own master.

adamaltman committed:
Add fast-route, and RouteMap class, and refactor MasterController.  Remove unused dependencies.

adamaltman commented:
Couldn't tag you from my repo's PR: https://github.com/adamaltman/masterclass-repo/pull/2  

I just made it.

adamaltman committed:
improve di config to set and get request service

adamaltman committed:
WIP: applied some improvements based on comments stop using global POST, GET, SERVER outside of the Request object.

- reduced usage of globals
- mostly finished Apc implementation
- changed the DataStore interface slightly
- temporarily worsened the Request object

adamaltman committed:
Improve the use of the router, and invert dependency into MasterController

adamaltman committed:
bug fixes: broke the router, save was replaced but usage of it wasn't (fixed now), make apc update safer

adamaltman committed:
Add the domain... it's a work in progress
